src/core/MOM_barotropic.F90

```diff
@@ -3413,6 +3415,44 @@ subroutine set_up_BT_OBC(OBC, eta, SpV_avg, BT_OBC, BT_Domain, G, GV, US, CS, MS
   call do_group_pass(BT_OBC%pass_h, BT_Domain)
   call do_group_pass(BT_OBC%pass_cg, BT_Domain)
 
+  ! Update IareaT_OBCmask so that nothing changes outside of the OBC (problem for interior OBCs only)
```

Comment:
The location and direction of the OBCs in the `OBC` type has already been set up at the point where `barotropic_init()` is called.  I think that if we passed `OBC` to `barotropic_init()`, we could move this block of code into `barotropic_init()`.  This would mean that we only need to add 1 halo update for IareaT_OBC_mask during initialization, rather than doing a halo update every time `btstep()` is called.